spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KeyValueSerdeResolver.java

```diff
@@ -140,6 +178,25 @@
 		return valueSerde;
 	}
 
+	public Serde<?> getOutboundValueSerde(ProducerProperties producerProperties,
+										KafkaStreamsProducerProperties kafkaStreamsProducerProperties, ResolvableType resolvableType) {
+		Serde<?> valueSerde;
+		try {
+			if (producerProperties.isUseNativeEncoding()) {
+				valueSerde = getValueSerde(
+						kafkaStreamsProducerProperties.getValueSerde(), resolvableType);
+			}
+			else {
+				valueSerde = Serdes.ByteArray();
+			}
+			valueSerde.configure(this.streamConfigGlobalProperties, false);
+		}
+		catch (ClassNotFoundException ex) {
```

Comment:
Given the possibility of CNFE I am assuming there is some class loading voodoo here, but I don't see which class loader is used. Just think if in this context it could be a problem

Reply:
ya, this CNFE is thrown by one of the Kafka classes and we are catching that and re-throw as an `IllegalStateException` with the original exception wrapped.